### ISSUE TITLE
How to Minify JSON

### ISSUE THREAD

**Issue by KyleMit** [terser/terser#540]
From the command line, the ReadMe indicates that you can set `--parse` to `expression` in order to:

> parse a single expression, rather than a program (for parsing json)

So if I have the following two files

**`statement.js`**
```js
var object = {"a" : "b"}
```

**`expression.json`**
```json
{"a" : "b"}
```

Then I should be able to run the following from the **command line**:

```bash
$ terser statement.js
// var object={a:"b"};
$ terser expression.json -p expression
// {a:"b"}
```

*However*, the same does not seem to work for the **node.js** API...

So parsing a JS statement works fine:

```js
const Terser = require("terser");

Terser.minify(`var object = {"a" : "b"}`)
// { code: "var object={a:"b"};"}
```

But looking at the [minify options](https://github.com/terser/terser#minify-options), the [`parse`](https://github.com/terser/terser#parse-options) option doesn't contain an `expression` option and the [`compress` options](https://github.com/terser/terser#compress-options) do, but it's intended to:

> Pass true to preserve completion values from terminal statements without return, e.g. in bookmarklets.

So neither of the following work to pass in options

```js
const Terser = require("terser");

Terser.minify(`{"a" : "b"}`, { compress: {expression: true}}) // Error: Unexpected token: punc (:)
Terser.minify(`{"a" : "b"}`, { parse: {expression: true}})    // TypeError: e.resolve_defines is not a function
```

And also, to be valid JSON, the object property would still need to retain the quotes around it.

Since the minification possible for JSON is really just removing whitespace, using `JSON.parse` and `JSON.stringify` should work just fine:

```js
JSON.stringify(JSON.parse(`{"a" : "b"}`))
```

Just wanted to make sure I wasn't missing anything obvious

**Comment by fabiosantoscode** [comment]
No, you're not missing anything obvious. It should be possible to use `parse: "expression"` in the node API as well. Thanks for the report!

Regarding minifying JSON (with a JSON output), as you say, everything that needs to be done is parsing and then stringifying. JSON.stringify produces no whitespace between tokens, therefore it's perfectly compact.

In node and the browser you can do what you specify in your last code example, and in the CLI you can use `jq . -c` if you have that amazing tool installed.

**Comment by jridgewell** [comment]
Pulling in https://github.com/terser/terser/issues/277:

> Not sure if bug or feature request. Depends on intended use I suppose.
> 
> The output of terser seems to always include a trailing semicolon: minify("(function() { return 5 })") => "(function{return 5});", which makes it hard to concat multiple results into one file as expressions: `"[" + result1 + "," + result2 + "]"`. I know I can do `{ output: { semicolons: false } }` to turn off ALL semicolons, but this seems like the wrong use for this option, and confusing for anyone looking at the code later.
> 
> My expectation would be that the "expression" option would also _output_ an expression. I understand that the actual purpose seems to be to keep final expressions.
> 
> That being said, I think it makes sense to be able to tell the generation that you are expecting an expression or statement/program on output, depending on how you are planning on using the generated output. I can of course merely remove the trailing semicolon, but this is a potentially expensive post-processing operation (result.substr(0, -1)), and also pretty brittle and difficult to understand, vs. a minify output option of `{ output: { expression: true } }` which would be more declarative and indicative of the intent, as well as faster than manipulating a potentially very large string. Such an option would mirror the parseExpression option in Babel.
